packages/material-ui/src/Hidden/HiddenCss.js

```diff
@@ -1,42 +1,85 @@
 import * as React from 'react';
+import clsx from 'clsx';
 import PropTypes from 'prop-types';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import capitalize from '../utils/capitalize';
-import withStyles from '../styles/withStyles';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import useTheme from '../styles/useTheme';
+import { getHiddenCssUtilityClass } from './hiddenCssClasses';
 
-const styles = (theme) => {
+const sizes = ['xl', 'lg', 'md', 'sm', 'xs'];
+
+const overridesResolver = (props, styles) => styles.root || {};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, breakpoints } = styleProps;
+
+  const slots = {
+    root: [
+      'root',
+      ...sizes.map((size) => breakpoints.includes(`${size}Up`) && `${size}Up`),
+      ...sizes.map((size) => breakpoints.includes(`${size}Down`) && `${size}Down`),
+      ...sizes.map(
+        (size) => breakpoints.includes(`only${capitalize(size)}`) && `only${capitalize(size)}`,
+      ),
```

Comment:
```suggestion
      'root',
      ...breakpoints,
```

Am I oversimplifying this? I think that breakpoints already contains all classes we want to add.